crates/oxc_linter/src/rules/eslint/sort_keys.rs

```diff
@@ -0,0 +1,1077 @@
+            .and_then(serde_json::Value::as_bool)
+            .unwrap_or(false);
+
+        Self(Box::new(SortKeysOptions {
+            sort_order,
+            case_sensitive,
+            natural,
+            min_keys,
+            allow_line_separated_groups,
+        }))
+    }
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        if let AstKind::ObjectExpression(dec) = node.kind() {
+            if dec.properties.len() < self.min_keys {
+                return;
+            }
+
+            let mut property_groups: Vec<Vec<String>> = vec![vec![]];
+
+            let source_text = ctx.semantic().source_text();
+
+            for (i, prop) in dec.properties.iter().enumerate() {
+                if let ObjectPropertyKind::SpreadProperty(_) = prop {
+                    property_groups.push(vec!["<ellipsis_group>".into()]);
+                    property_groups.push(vec![]);
+                    continue;
+                }
+                let key = match prop.prop_name() {
+                    Some((name, _)) => name,
+                    // FIXME: prop_name is currently unset for computed properties, short hand properties, and function declarations.
+                    None => extract_property_key(prop.span().source_text(source_text)),
+                };
+
+                if i != dec.properties.len() - 1 && self.allow_line_separated_groups {
+                    let text_between = extract_text_between_spans(source_text, prop.span(), dec.properties[i + 1].span());
+                    if text_between.contains("\n\n") {
+                        property_groups.last_mut().unwrap().push(key.into());
+                        property_groups.push(vec!["<linebreak_group>".into()]);
+                        property_groups.push(vec![]);
+                        continue;
+                    }
+                }
+                property_groups.last_mut().unwrap().push(key.into());
+            }
+
+            if !self.case_sensitive {
+                for group in &mut property_groups {
+                    *group = group.iter()
+                        .map(|s| s.to_lowercase())
+                        .collect::<Vec<String>>();
+                }
+            }
+
+            let mut sorted_property_groups = property_groups.clone();
+            for (i, group) in property_groups.iter().enumerate() {
+                let mut sorted = group.clone();
+
+                if self.natural {
+                    natural_sort(&mut sorted);
+                } else {
+                    alphanumeric_sort(&mut sorted);
+                }
+
+                if self.sort_order == SortOrder::Desc {
+                    sorted.reverse();
+                }
+
+                sorted_property_groups[i] = sorted;
+            }
+
+            let is_sorted = all(property_groups.iter().zip(&sorted_property_groups), |(a, b)| a == b);
+
+            if !is_sorted {
+                ctx.diagnostic(
```

Comment:
Use a diagnostic factory function. See other rules for examples.

Reply:
👍🏼